### ISSUE TITLE
Add basic ORM around storage

### ISSUE THREAD

**Issue by ethanfrey** [CosmWasm/cosmwasm-go#60]
Depends on #58 

Something like `cosmwasm-storage` or `cw-storage-plus`.

We may be able to draw inspiration from the [ORM in Weave](https://github.com/iov-one/weave/tree/master/orm) but simpler without reflect and without Protobuf codecs.

* Expose typed helpers for storage
* Support for singletons and "maps"/"buckets"
* *Maybe* basic secondary index support (as a separate issue/PR)

Here is an example of current usage:

```go
// this is what we store
type State struct {
	Verifier    string `json:"VERIFIER"`
	Beneficiary string `json:"BENEFICIARY"`
	Funder      string `json:"FUNDER"`
}

var StateKey = []byte("config")

func LoadState(storage std.Storage) (*State, error) {
	data, err := storage.Get(StateKey)
	if err != nil {
		return nil, err
	}

	var state State
	err = state.UnmarshalJSON(data)
	if err != nil {
		return nil, err
	}
	return &state, nil
}

func SaveState(storage std.Storage, state *State) error {
	bz, err := state.MarshalJSON()
	if err != nil {
		return err
	}
	// TODO: storage.Set should not return error
	return storage.Set(StateKey, bz)
}
```

Ideally this would look something like:

```go
// this is what we store
type State struct {
	Verifier    string `json:"VERIFIER"`
	Beneficiary string `json:"BENEFICIARY"`
	Funder      string `json:"FUNDER"`
}

// not sure if we can use types here given reflect limits
var stateBucket = NewBucket("config", &State{})

// example usage
func foo() {
  var State state
  stateBucket.Load(deps.Storage, &state)
  state.Verifier = "foo"
  stateBucket.Save(deps.Storage, state)
}
```
